### PR TITLE
YUBA-86 Add at_boost_url lib to enable usage or boost::urls

### PR DESCRIPTION
**Pull request by MattMacGregor**
For YUBA-86 we are taking app specified keywords attached to the stream metadata, and submitting them to Deepgram as "Keywords".

Because these are passed as URL arguments as part of the websocket endpoint, I would like some safety regarding URL reserved characters. Boost has a helpful URL encode function, but it's source file (libs/url/src/encoding_opts.cpp) wasn't included in any of the existing libraries we have in the CMake file. So this just adds a simple library with that URL encoding function.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -256,3 +256,16 @@ set_target_properties(at_boost_program_options
    PROPERTIES COMPILE_FLAGS "-Wno-overloaded-virtual"
 )
 
+##
+## URLS
+##
+set(BOOST_URL_ENCODE_SOURCES
+  libs/url/src/encoding_opts.cpp
```
Comment (artasano) [inline CMakeLists.txt:265]: boost/libs/url/src has more `cpp`s. Is just adding this one enough?

Comment (MattMacGregor) [reply CMakeLists.txt:265]: It is for using the `boost::urls::encode` function, which is all I need for Yuba.

I figured in the future if we ever need the rest of the Boost URLs library functionality we could just add them then.
But I'm open to opinions, doesn't really matter all that much

Comment (artasano) [reply CMakeLists.txt:265]: I'd vote for adding all now. Is build time a concern?

Comment (MattMacGregor) [reply CMakeLists.txt:265]: Nope not a concern, I'll add them all then, and update the name to be `at_boost_url`

**Comment by MattMacGregor** [comment]
Alright updated it to build the whole url lib. Did a test build with yuba and everything compiled and linked up fine.

File: CMakeLists.txt (hunk 1 above)
Comment (MattMacGregor) [reply CMakeLists.txt:265]: Updated @artasano

**Review by artasano — APPROVED** [review]

**Pull request merged**